test/functional/test_framework/mininode.py

```diff
@@ -647,3 +648,11 @@ def on_inv(self, message):
     def get_invs(self):
         with mininode_lock:
             return list(self.tx_invs_received.keys())
+
+    def wait_for_broadcast(self, num_txns):
+        # Waits for num_txns transactions to complete initial broadcast.
+        # Wait until num_txns invs have been received (and getdatas sent).
+        wait_until(lambda: len(self.tx_invs_received.keys()) == num_txns)
```

Comment:
```suggestion
        self.wait_until(lambda: set(self.get_invs()) == set(txns))
```

* self.wait_until because we need the scaling factor
* self.get_inv because we need to take the lock
* set because we want to check the txids

Reply:
This makes sense. 

It seems like a `P2PInterface`s don't have a `self.wait_until` function. I get a `object has no attribute 'wait_until'` error. The classes in `mininode.py` use `wait_until` with the `mininode_lock`.  I'm not familiar with what you mean by scaling factor, but is there something else I could do?

Reply:
Are you sure you are on current master? I am seeing this function locally:

```
$ git grep 'def wait_until'
test/functional/test_framework/mininode.py:    def wait_until(self, test_function, timeout):